Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.cpp

```diff
@@ -239,6 +252,49 @@ void ImageBufferIOSurfaceBackend::ensureNativeImagesHaveCopiedBackingStore()
     flushContext();
 }
 
+void* ImageBufferIOSurfaceBackend::updateBitmapCacheForRead()
+{
+    if (!m_bitmapCacheLock.has_value()) {
+        flushContext();
+        m_bitmapCacheLock = IOSurface::Locker { *m_surface, IOSurface::Locker::AccessMode::ReadOnly };
+    }
+    return m_bitmapCacheLock->surfaceBaseAddress();
+}
+
+CGImageRef ImageBufferIOSurfaceBackend::updateBitmapCacheImage()
+{
+    if (m_bitmapCacheImage)
+        return m_bitmapCacheImage.get();
+    void* data = updateBitmapCacheForRead();
+    if (!data)
+        return nullptr;
+    auto configuration = m_surface->bitmapConfiguration();
+    auto size = m_surface->size();
+    // FIXME: we can cache also the context and just use set data operation, in later commits.
+    auto context = adoptCF(CGBitmapContextCreate(data, size.width(), size.height(), configuration.bitsPerComponent, m_surface->bytesPerRow(), m_surface->colorSpace().platformColorSpace(), configuration.bitmapInfo));
+    if (!context)
+        return nullptr;
+    m_bitmapCacheImage = adoptCF(CGBitmapContextCreateImage(context.get()));
+    return m_bitmapCacheImage.get();
+}
+
+void* ImageBufferIOSurfaceBackend::updateBitmapCacheForWrite()
+{
+    releaseBitmapCache();
```

Comment:
Why do we need to call `releaseBitmapCache()` here? Have we ever need to cache the lock of the CGImage after they are used?

Reply:
This one relates to the one comment  of upcoming scenario where, after we track draws, the cache can be left open.